genai-perf/genai_perf/config/generate/perf_analyzer_config.py

```diff
@@ -34,14 +89,36 @@ def __init__(
         config: ConfigCommand,
         model_objective_parameters: ModelObjectiveParameters,
         model_name: ModelName,
+        args: Namespace = Namespace(),
+        extra_args: Optional[List[str]] = None,
     ):
         self._model_name = model_name
+        self._args = deepcopy(args)
+        self._set_options_based_on_cli(args, extra_args)
         self._set_options_based_on_config(config)
         self._set_options_based_on_objective(model_objective_parameters)
+        self._set_artifact_paths()
 
     ###########################################################################
     # Set Options Methods
     ###########################################################################
+    def _set_options_based_on_cli(
+        self, args: Namespace, extra_args: Optional[List[str]] = None
+    ) -> None:
+        self._cli_args = []
+
+        # When restoring from a checkpoint there won't be any args
+        if not hasattr(self._args, "subcommand"):
+            return
+
+        self._cli_args += self._add_required_args(args)
+        self._cli_args += Profiler.add_protocol_args(args)
+        self._cli_args += Profiler.add_inference_load_args(args)
+        self._cli_args += self._add_misc_args(args)
+
+        if extra_args:
+            self._cli_args += self._add_extra_args(extra_args)
```

Comment:
The check if `extra_args` is empty/null is done on both before and within the call to `_add_extra_args`, I think we can safely remove this check (or vice versa).

Reply:
Good catch. I've removed this redundancy.